### PR TITLE
Types.ReactSvgRadarChart. Adding react-svg-radar-chart typings.

### PR DESCRIPTION
**Pull request by Luksys5**
Hey,
I'm the creator of typings. 
In case you have any trouble with it then feel free to contact me.

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

If removing a declaration:
- [x] Add it to `notNeededPackages.json`.

Cheers!

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -5247,6 +5247,7 @@
 /types/react-stripe-elements/                           @dan-j @santiagodoldan @sonnysangha @9y5 @thchia @yhnavein @virzak @remotealex @bombek92 @hirochachacha @paustint @mastacheata
 /types/react-svg-inline/                                @kiyopikko
 /types/react-svg-pan-zoom/                              @huy-nguyen
+/types/react-svg-radar-chart/                           @luksys5
```
Comment (peterblazejewicz) [inline .github/CODEOWNERS:5250]: I believe this is automatic, not changed manually via PR
```suggestion
```

File: types/react-svg-radar-chart/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+// Type definitions for react-svg-radar-chart 1.2
+// Project: https://github.com/Spyna/react-svg-radar-chart
+// Definitions by: Lukas Tutkus <https://github.com/luksys5>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Typescript Version 3.5
+// Minimum TypeScript Version: 3.0
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:6]: Why both (both works btw)? Is the minimum 3.5?
```suggestion
```

File: types/react-svg-radar-chart/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+export interface ChartData {
+    data: {
+        [value: string]: number;
+    };
+    meta: { color: string };
+}
+
+export interface ChartOptionsProps {
+    size: number;
+    axes: boolean; // show axes?
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:19]: This inline comment does not work well with intelisense
```suggestion
    /** show axes */
    axes: boolean;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    scales: number; // show scale circles?
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:20]:
```suggestion
    /** show scale circles */
    scales: number;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    captions: boolean; // show captions?
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:21]:
```suggestion
    /** show captions */
    captions: boolean;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    captionMargin: number;
+    dots: false; // show dots?
+    zoomDistance: 1.2; // where on the axes are the captions?
+    setViewBox: (options: ChartOptionsProps) => number; // custom viewBox ?
+    smoothing: (points: ReadonlyArray<[]>) => string; // shape smoothing function
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:26]:
```suggestion
    /**
     * show dots
     * @default false
     */
    dots?: boolean;
    /**
     * where on the axes are the captions
     * @default 1.2
     */    
    zoomDistance?: number;
    /** custom viewBox */
    setViewBox?: (options: ChartOptionsProps) => number;
    /** shape smoothing function */
    smoothing?: (points: ReadonlyArray<[]>) => string;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    axisProps: () => { className: string };
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:27]:
```suggestion
    axisProps: () => { className: 'axis' | string };
```

File: types/react-svg-radar-chart/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    scaleProps: () => {
+        className: string;
+        fill: string;
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:30]:
```suggestion
        className: 'scale' | string;
        fill: 'none' | string;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    };
+    shapeProps: () => { className: string };
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:32]:
```suggestion
    shapeProps: () => { className: 'shape' | string };
```

File: types/react-svg-radar-chart/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    captionProps: () => {
+        className: string;
+        textAnchor: string;
+        fontSize: number;
+        fontFamily: string;
+    };
+    dotProps: () => {
+        className: string;
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:40]:
```suggestion
        className: 'dot' | string;
```

File: types/react-svg-radar-chart/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    scaleProps: () => {
+        className: string;
+        fill: string;
+    };
+    shapeProps: () => { className: string };
+    captionProps: () => {
+        className: string;
+        textAnchor: string;
+        fontSize: number;
+        fontFamily: string;
```
Comment (peterblazejewicz) [inline types/react-svg-radar-chart/index.d.ts:37]:
```suggestion
        className: 'caption' | string;
        textAnchor: 'middle' | string;
        fontSize: number;
        fontFamily: 'sans-serif' | string;
```

**Comment by peterblazejewicz** [comment]
@Luksys5 
those lines:
```ts
fontFamily?: 'sans-serif' | string
```
works as self-documenting code (at compile time it would be transformaed just to `string | undefined`.
You could also just provide correct JSDoc with `@default` values simply, up to you.

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
- IMO code owners is automatic, remove that change
- review the JSDoc @defult tag values
- review properties being optional, with default values that could be added via JSDoc
Looks good, minor corrections required

File: types/react-svg-radar-chart/index.d.ts (hunk 2 above)
Comment (Luksys5) [reply types/react-svg-radar-chart/index.d.ts:6]: Actually I've only tested on 3.5 so I should remove the minimum. Thought that it was not that different

File: .github/CODEOWNERS (hunk 1 above)
Comment (Luksys5) [reply .github/CODEOWNERS:5250]: Will remove then this one. Thanks

File: types/react-svg-radar-chart/index.d.ts (hunk 3 above)
Comment (Luksys5) [reply types/react-svg-radar-chart/index.d.ts:19]: Will remove it

**Comment by elibarzilay** [comment]
@Luksys5, @peterblazejewicz: Is this ready to be merged?  (Looks like there are a few unresolved comments.)

**Comment by peterblazejewicz** [comment]
@elibarzilay 
I could accept stright away, there is a single minor problem with the DT header really (other can be amended later by anyone):
```ts
// Typescript Version 3.5
// Minimum TypeScript Version: 3.0
```
This is not catched by linter (two different TS versions reqruied in header). Not sure how to approach that (no update on this from OP)

**Comment by elibarzilay** [comment]
@peterblazejewicz  OK, I'll merge it as is for now...

**Comment by peterblazejewicz** [comment]
@Luksys5 when you have time, please revisit topics discussed! Thanks!

**Comment by Luksys5** [comment]
@peterblazejewicz Yeah actually I forgot to push my changes regarding the typescript version and other resolved things. 
I will create another pull request for changing that.
Sorry about the delay

**Pull request merged**